src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/FilterRoleAssignmentsImpl.java

```diff
@@ -0,0 +1,139 @@
+package uk.gov.hmcts.ccd.domain.service.accessprofile.filter;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.ccd.data.casedetails.SecurityClassification;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignments;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.std.CaseDataContent;
+import uk.gov.hmcts.ccd.domain.service.AccessControl;
+import uk.gov.hmcts.ccd.domain.service.common.SecurityClassificationUtils;
+
+import static uk.gov.hmcts.ccd.domain.service.common.SecurityClassificationUtils.caseHasClassificationEqualOrLowerThan;
+
+@Slf4j
+@Component
+public class FilterRoleAssignmentsImpl implements FilterRoleAssignments, AccessControl {
+
+    @Override
+    public RoleAssignments filter(RoleAssignments roleAssignments, CaseDetails caseDetails) {
+        log.info("Filter role assignments for case {}", caseDetails.getReference());
+
+        List<RoleAssignment> roleAssignmentList = roleAssignments.getRoleAssignments().stream()
+            .filter(roleAssignment -> validateStartAndEndDate(roleAssignment))
+            .filter(roleAssignment -> validateSecurityValidation(caseDetails, roleAssignment))
+            .filter(roleAssignment -> validateCaseId(caseDetails, roleAssignment))
+            .filter(roleAssignment -> validateJurisdiction(caseDetails, roleAssignment))
+            .filter(roleAssignment -> validateLocation(roleAssignment))
+            .filter(roleAssignment -> validateRegion(roleAssignment))
+            .collect(Collectors.toList());
+
+        return RoleAssignments.builder().roleAssignments(roleAssignmentList).build();
```

Comment:
perhaps simpler just to add a constructor
`return new RoleAssignments(roleAssignmentList)`
?